### DISCUSSION TITLE
How can I specify an Airflow version when using the Helm chart

### DISCUSSION THREAD

**Discussion by robteeuwen** [apache/airflow#26804] [Q&A]
Hello! 

This may be a stupid question, but I can't figure out how to specify a different version of Airflow in my deployment that I installed using the official Helm chart. 

I'm running Airflow in Kubernetes on Azure. I installed it there by using the Helm chart. I get my DAGs inside by baking them into the image according to the documentation (https://airflow.apache.org/docs/helm-chart/stable/manage-dags-files.html#bake-dags-in-docker-image). I noticed that the version updates automatically, and is currently at 2.4.0, which normally would be great. However, I've been having some issues due to the bug regarding the webserver being unable to fetch logs from the worker (https://github.com/apache/airflow/discussions/26490). This issue was solved in 2.4.1, which isn't yet the latest release (I can only find 2.4.1.rc1). So, I would like to force my deployment to use this tag, or use 2.3.4 instead, but I can't figure out how to do it. 

I've tried several things: 
- I changed my pipfile (I use pipenv for dependency management) to `apache-airflow = "2.3.4"`, but both webserver and the python package are still 2.4.0
- In my Dockerfile, I changed `FROM apache/airflow` into `FROM apache/airflow:2.3.4`, but nothing changes (webserver and python package are still 2.4.0), even though, the output in my devop pipeline states: `2.3.4: Pulling from apache/airflow` when I build the image with my DAGs in it. This image is the one referenced in the images.airflow section of the Helm chart. 

My dockerfile looks like this: 

```
FROM apache/airflow:2.3.4

RUN pip install pipenv
COPY ./Pipfile ${AIRFLOW_HOME}/Pipfile
COPY ./Pipfile.lock ${AIRFLOW_HOME}/Pipfile.lock
RUN pipenv install --deploy --system --skip-lock 

COPY ./dags/ ${AIRFLOW_HOME}/dags/
```

Something else I don't understand is that in the Helm repo, application version is listed as 2.3.0 (https://artifacthub.io/packages/helm/apache-airflow/airflow), which corresponds to the output I'm getting from Helm after upgrading in my devops pipeline (Thank you for installing Apache Airflow 2.3.0!) and the value I see when running `helm list`. Does this refer to the version of the Helm chart? Why is it not the same as the application version? I also noticed that values.yaml

**Comment by potiuk** [comment]
You should use `images.airflow` parameters (see https://airflow.apache.org/docs/helm-chart/stable/parameters-ref.html#common using standard way you set parameters in Helm chart. Look for example in https://helm.sh/docs/chart_template_guide/values_files/ where the concept of Helm Chart values is described. 

You can see an example of overriding values by ``--set`` command in https://airflow.apache.org/docs/helm-chart/stable/manage-dags-files.html
